### PR TITLE
Dialect methods implementation

### PR DESCRIPTION
**Pull request by mrbungie**
Implements #2. With this the dialect has enough functionality to be used  with Airbnb's Caravel.

### CODE REVIEW

File: sqlalchemy_teradata/dialect.py [hunk 1]
```diff
@@ -116,4 +116,154 @@ def get_view_names(self, connection, schema=None, **kw):
...
+        stmt = select([column('ColumnName'), column('IndexName')], from_obj=[text('dbc.Indices')]) \
+            .where(and_(text('DatabaseName = :schema'),
+                        text('TableName=:table'),
+                        text('IndexType=:indextype'))) \
+            .order_by(asc(column('IndexName')))
+
+        # U for Unique
+        res = connection.execute(stmt, schema=schema, table=table_name, indextype='U').fetchall()
+
+        def grouper(fk_row):
```
Comment (sandan) [inline sqlalchemy_teradata/dialect.py]: `grouper` never gets used in `get_unique_constraints`

Comment (mrbungie) [inline sqlalchemy_teradata/dialect.py]: Fixed

File: sqlalchemy_teradata/dialect.py [hunk 2]
```diff
@@ -116,4 +116,154 @@ def get_view_names(self, connection, schema=None, **kw):
...
+            return {
+                'name': fk_row.IndexName,
+            }
+
+        unique_constraints = list()
```
Comment (sandan) [inline sqlalchemy_teradata/dialect.py]: Ideally we'd like to push as much of the SQL computation to the database so processing is kept simple. I presume that is what your TODO comments are referring to in `get_foreign_keys` and  `get_indexes`.

This is a good implementation for now. Thanks!

Comment (mrbungie) [inline sqlalchemy_teradata/dialect.py]: Yes that's the reason of those TODOs.

Glad you appreciated it! I will continue pushing PRs as long as I find something I can add/fix/improve.

**Pull request merged**